### PR TITLE
Use cache for integration tests

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
We use a bunch of golang to install dependencies in the `github-actions-setup` script, means we should use a cache to speedup the CI.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -20,13 +20,23 @@ jobs:
       - uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
+      - run: sudo chown -R $(id -u):$(id -g) /root
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
```
Comment (sohankunkerkar) [inline .github/workflows/integration.yml:27]: I think a somewhat similar change has been removed recently in [this](https://github.com/cri-o/cri-o/commit/aa86e94f04c806b26abfafb323a94ab470de66d1) commit.

**Comment by haircommander** [comment]
cc @kolyshkin

File: .github/workflows/integration.yml (hunk 1 above)
Comment (kolyshkin) [reply .github/workflows/integration.yml:27]: actions/setup-go@v4 caches a few directories -- `$(go env GOMODCACHE)` and `$(go env GOCACHE)`, to be precise, those are usually `~/go/pkg/mod` and `~/.cache/go-build`.

I think we should build all sources and tests as non-root (and I don't see any cases where we run `go build` or `go test` as root, maybe I'm mistaken?), so the above caching should be sufficient.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -20,13 +20,23 @@ jobs:
...
+            ~/.cache/go-build
+            /root/go/pkg/mod
+            /root/.cache/go-build
+          key: go-integration-${{ hashFiles('scripts/versions') }}-${{ hashFiles('scripts/github-actions-setup') }}
```
Comment (kolyshkin) [inline .github/workflows/integration.yml:31]: and, if you want hashes of these files to be used as cache key input, there's a `cache-dependency-path` parameter described in actions/setup-go README.md (if unset, it looks for `go.sum` file in `$GITHUB_WORKSPACE`)

Something like this should work:

```yml
    cache-dependency-path: |
        ./go.sum
        ./scripts-versions
        ./scripts-github-actions-setup
```

OTOH, I think that changing a version of a tool we download should not invalidate go cache, so this looks like not being necessary.

**Comment by saschagrunert** [comment]
I think we can solve this by changing the script.

**Pull request closed without merging**